ui/src/main/java/org/jtaccuino/core/ui/extensions/DisplayExtension.java

```diff
@@ -130,241 +113,62 @@ public void display(Object object, Consumer<Integer> counterConsumer) {
-                        return valueProp;
-                    });
-                    tv.getColumns().add(tcKey);
-                    if (me.getValue().getClass().isRecord()) {
-                        tv.getColumns().addAll(tableColumnsFromRecordComponentsForMapValues(me.getValue().getClass()));
-                    } else {
-                        var tcValue = new TableColumn<Map.Entry<Object, Object>, String>("Value");
-                        tcValue.setCellValueFactory(o -> {
-                            var valueProp = new SimpleStringProperty();
-                            if (null != o) {
-                                valueProp.set(String.valueOf(o.getValue().getValue()));
-                            }
-                            return valueProp;
-                        });
-                        tv.getColumns().add(tcValue);
-                    }
-                    s.stream().map(o -> (Map.Entry<Object, Object>) o).forEach(e -> tv.getItems().add(e));
-                    tv.getColumns().forEach(c -> c.setPrefWidth(COLUMN_PREF_WIDTH));
-                    tv.setMaxHeight(CELL_MAX_HEIGHT);
-                    yield tv;
-                } else {
-                    var observableList = FXCollections.observableList(s.stream().map(String::valueOf).toList());
-                    var lv = new ListView<String>(observableList);
-                    lv.setMaxHeight(CELL_MAX_HEIGHT);
-                    yield lv;
-                }
-            }
-            case Object[] a -> convertArrayToNode(a);
-            case double[] a -> convertArrayToListView(Arrays.stream(a).mapToObj(Double::valueOf).toArray());
-            case int[] a -> convertArrayToListView(Arrays.stream(a).mapToObj(Integer::valueOf).toArray());
-            case long[] a -> convertArrayToListView(Arrays.stream(a).mapToObj(Long::valueOf).toArray());
-            default -> {
-                var label = new Label("No automatic conversion for type " + object.getClass() + " + to javafx.scene.Node found!");
-                yield label;
-            }
-        };
-    }
+    private long lastEffect;
 
-    protected <T> Node convertArrayToNode(T[] a) {
-        if (a.length > 0 && null != a[0]) {
-            if (a[0].getClass().isRecord()) {
-                var tv = new TableView<Object>();
-                tv.getColumns().addAll(tableColumnsFromRecordComponents(a[0].getClass()));
-                tv.getItems().addAll(a);
-                tv.getColumns().forEach(c -> c.setPrefWidth(COLUMN_PREF_WIDTH));
-                tv.setMaxHeight(CELL_MAX_HEIGHT);
-                return tv;
-            }
-        }
-        var observableList = FXCollections.observableList(Arrays.stream(a).map(String::valueOf).toList());
-        var lv = new ListView<String>(observableList);
-        lv.setMaxHeight(CELL_MAX_HEIGHT);
-        return lv;
-    }
+    private Comparator<ServiceLoader.Provider<NodeRenderer>> NODE_RENDERER_COMPARATOR = Comparator.comparing(
```

Comment:
Any reason why this is a regular instance field and not static
```suggestion
    private static final Comparator<ServiceLoader.Provider<NodeRenderer>> NODE_RENDERER_COMPARATOR = Comparator.comparing(
```